src/dotnet/AzureAI.Proxy/ReverseProxy/ThrottlingHealthPolicy.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Yarp.ReverseProxy.Health;
+using Yarp.ReverseProxy.Model;
+
+namespace AzureAI.Proxy.ReverseProxy;
+
+public class ThrottlingHealthPolicy : IPassiveHealthCheckPolicy
+{
+    public static string ThrottlingPolicyName = "ThrottlingPolicy";
+    private readonly IDestinationHealthUpdater _healthUpdater;
+
+    public ThrottlingHealthPolicy(IDestinationHealthUpdater healthUpdater)
+    {
+        _healthUpdater = healthUpdater;
+    }
+
+    public string Name => ThrottlingPolicyName;
+
+    public void RequestProxied(HttpContext context, ClusterState cluster, DestinationState destination)
+    {
+        var headers = context.Response.Headers;
+
+        if (context.Response.StatusCode is 429 or >= 500)
+        {
+            var retryAfterSeconds = 10;
```

Comment:
When will this be triggered? Should we lower this value?

Reply:
this is the initial value and will be updated based on the headers. I'll leave this